SignalMessaging/attachments/AttachmentApprovalViewController.swift

```diff
@@ -624,58 +628,41 @@ class CaptioningToolbar: UIView, UITextViewDelegate {
         // Increase hit area of send button
         sendButton.contentEdgeInsets = UIEdgeInsets(top: 6, left: 8, bottom: 6, right: 8)
 
-        addSubview(bottomGradient)
-        addSubview(sendButton)
-        addSubview(textView)
-
-        sendButton.sizeToFit()
-    }
-
-    func didTapSend() {
-        self.captioningToolbarDelegate?.captioningToolbarDidTapSend(self, captionText: self.textView.text)
-    }
-
-    // MARK: - UIView Overrides
-
-    // We do progammatic layout, explicitly computing and setting frames since autoLayout does
```

Comment:
Turns our you don't need to do programmatic layout if you properly set up auto resizing. I think the autolayout code is a lot easier to reason about.

Reply:
Awesome.